deviseの導入、モデル・コントローラーの作成、viewファイルの追加

norika69 opened the pull request:
*what
deviseの導入とコントローラー、モデルの作成
viewファイルの追加
*why
ログイン機能実装のため


フロント実装終了のタイミングでpullriqし忘れてしまい、
フロント実装とdevise導入が同じpullriqになってしまっております。
フロント実装は終了しておりますので、devise導入部分からレビューをお願いできたらと思います。

norika69 committed:
application.cssからapplication.scssに変換

cssをsassへ変換
_riset.scssの作成

norika69 committed:
font-awesomeインストール

gemによるfont-awesomeというフォントをrailsに導入。
キーワードで検索した場合でもコードのようなアイコンが使用できるようになる。

norika69 committed:
messagesコントローラーの作成

トップページを表示させるため
rails g controller messages
rake db:create
indexアクションの作成

norika69 committed:
messageコントローラーのルーティング設定

resourcesでonly: :indexにする

norika69 committed:
font-awesomeの変更

font-awesome-railsにしてしまっていたのでsassで使えるようにfont-awesome-sassに変更

norika69 committed:
application.scssの指定ファイル順番変更

上から順に読まれるため最上部にreset.scssを指定。

norika69 committed:
チャット画面左側のscssを指定

norika69 committed:
mixin clear fixの設定

mixingにclear fixを設定してfloat解除

norika69 committed:
index/scss

indexビューのsassを設定

norika69 committed:
index/scss

indexのsassを設定

norika69 committed:
add devise

deviseの導入

norika69 committed:
create message model

messageモデルの作成

norika69 committed:
add column to message table

massage tableへのカラム追加
body
image
group_id
user_id

norika69 committed:
setting root_path

ルートパスの設定
messages#index

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,221 @@
+  &__left {
+    background-color: #2f3e51;
+    width: 300px;
+    height: 100vh;
```
細かいですが、widthとheightの順序は統一しましょう！

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,221 @@
+      @include clearfix();
+      height: 101px;
+      background-color: #FFFFFF;
+      border-bottom: 1px solid #eeeeee;
```
カラーコードの小文字大文字も統一しましょう！
また、カラーコードが1文字で構成されている場合は`#eee`のように記述するのが一般的なので、他の部分も修正お願い致します。

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,221 @@
+      height: 101px;
+      background-color: #FFFFFF;
+      border-bottom: 1px solid #eeeeee;
+      padding: 0 40px 0 40px;
```
上下と左右がそれぞれ同一の値の場合は`padding: 上下 左右;`という書き方ができます

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,221 @@
+      h2 {
+        font-size: 20px;
+        color: #333333;
+        margin: 20px 0 0 0;
```
この場合は`margin-top`を使用するべきですね

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,221 @@
+        float: right;
+        .editbutton {
+          margin-top:28px;
+          border: 2px #38AEF0 solid;
```
`#38AEF0`こちらのカラーコードは複数回使用しているので変数化しましょう！
変数の使用方法がわからない場合は`SASS 変数`で調べるとよろしいかと思います

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,221 @@
+          padding: 40px 0 40px 40px;
+          list-style: none;
+          // margin-bottom: 40px;
+          // margin-top: 40px;
```
必要なければコメントアウトは消しましょう！
残してPRを出す場合は残している理由を書くようにしましょう

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,221 @@
+        .editbutton {
+          margin-top:28px;
+          border: 2px #38AEF0 solid;
+          background-color : white;
```
`:`の前にスペース入ってしまっていますね

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,221 @@
+              float: left;
+              font-size: 16px;
+              font-weight: bold;
+              color: 434A54;
```
カラーコードになっていないです

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,221 @@
+        @include clearfix();
+        max-width: 100%;
+        form {
+          margin: 20px 0 20px 0;
```
marginも同様に上下と左右で指定しましょう

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,221 @@
+          margin-right: 10px;
+          background-color: white;
+          position: relative;
+          display: inline-block;
```
プロパティの順序がバラバラなので、できるだけ統一しましょう！

madana-i commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,221 @@
+          position: relative;
+          display: inline-block;
+        }
+          lavel {
```
`label`の間違いでしょうか？

madana-i commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,44 @@
+        .chat__mail__footer__form__body
+          %form{:action => "#", :method => "post"}
+            %textarea{:placeholder => "type a massage" , :rows => "1", :id => "message-body"}
+            %lavel.chat__main__footer__form__body__lavel
```
lavelではなくlabelですね！

madana-i commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,44 @@
+      .chat__main__footer__form
+        .chat__mail__footer__form__body
+          %form{:action => "#", :method => "post"}
+            %textarea{:placeholder => "type a massage" , :rows => "1", :id => "message-body"}
```
ハッシュロケット(=>)の書き方はあまり使われないのでシンボル型で書くようにしましょう！
また、式展開を使用しない場合はシングルクォーテーションを使用しましょう！

norika69 committed:
refactoring index view file

haml,scssの修正

Hiroaki-Shimohira commented:
viewは問題なさそうです！
deviseの途中の部分を実装してからもう一度プルリク出してください。

norika69 committed:
create devise users model

deviseコマンドによるusersモデルの作成
rails g devise users

norika69 committed:
create devise users model2

deviseコマンドによるusersモデルの作成
rails g devise user

コミット漏れです。

norika69 committed:
change column constraint of message table

messageテーブルのカラム作成
カラム,型,制約
body,text,null_false
image,string,
group_id,integer,foreign_key
user_id,integer,foreign_key

norika69 committed:
create users table

users テーブルの作成
rake db:migrate

norika69 committed:
create devise views

devise関連のビューの作成
rails g device:views

hand12 commented on app/assets/javascripts/messages.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
使っていないのであれば削除してしまいましょう！
のちに使うという場合、その時に作成してコミットしましょう！

hand12 commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,220 @@
+          padding: 40px 0 40px 40px;
+          list-style: none;
+          // margin-bottom: 40px;
+          // margin-top: 40px;
```
どこの部分かを残しておくコメントアウトはわかりやすくていいですね！
ただ、使っていないコメントアウトは自分で読む際はわかるかもしれないのですが、他の人が読む際にわかりづらくなってしまうので消すようにしましょう！

hand12 commented on app/models/message.rb:
```diff
@@ -0,0 +1,2 @@
+class Message < ApplicationRecord
+end
```
今回のこのブランチとは関係ないコミットはやめましょう！
また、モデルを作成したときはバリデーションも一緒につけるようにして下さい。
今回はこのままで大丈夫ですので次回から気を付けましょう！

hand12 commented on app/views/devise/confirmations/new.html.erb:
```diff
@@ -0,0 +1,16 @@
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
hamlに直してコミットしましょう！

hand12 commented on app/views/devise/registrations/new.html.erb:
```diff
@@ -0,0 +1,29 @@
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
こちらはhamlのファイルがすでに存在していますのでいらないですね。
hamlに書き換える際は、もとのerbを全てhamlに置き換えるというコマンドがあるのでぜひ使ってみてください！

hand12 commented on db/migrate/20161120171900_create_messages.rb:
```diff
@@ -0,0 +1,11 @@
+class CreateMessages < ActiveRecord::Migration[5.0]
+  def change
+    create_table :messages do |t|
+      t.text  :body, null: false
```
シチュエーションを想定すると、画像だけ投稿してbodyがない場合もあると思います。
なので、null falseをつけてしまうとnullを許可されていない = からではレコードを保存できないので、そこを考えてみましょう！

hand12 commented on db/migrate/20161120171900_create_messages.rb:
```diff
@@ -0,0 +1,11 @@
+      t.text  :body, null: false
+      t.string  :image
+      t.integer  :group_id, foreign_key: true
+      t.integer  :user_id, foreign_key: true
```
外部キーとの連結は、integer型でもできるのですが、参照型というのもあるので調べてみてください！

hand12 commented on test/controllers/messages_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
テストは自動で作られるとは思いますが、今回のブランチでは使わないのでコミットしないようにしましょう「！

hand12 reviewed (commented):
testと書いてある部分は全ていらないです。また、erbファイルもいらないので、いらないファイルはコミットしないようにしましょう！

norika69 committed:
drop files

不要なファイルの削除
（text,coffee,helper)
erb→hamlへの変換でerbファイル削除
rake haml:erb2haml コマンド

可読性を高めるため

norika69 committed:
setting validation to user model

userモデルへバリテーション設定
emailカラムの空欄はfalse

norika69 committed:
create group model

groupモデルの作成
外部キーを持たせるため。

norika69 committed:
setting foreign_key

#外部キーの設定
一度ロールバックしてmessageテーブルのマイグレーションファイルを編集し直した。
integer => references

hand12 commented on test/fixtures/groups.yml:
```diff
@@ -0,0 +1,11 @@
+# column: value
+#
+two: {}
+# column: value
```
testファイルは消してしまいましょう！

hand12 commented on test/models/group_test.rb:
```diff
@@ -0,0 +1,7 @@
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
こちらもですね！

hand12 reviewed (commented):
不要なファイルのコミットがありますので、そこだけ確認して修正したらマージして次に進みましょう！

norika69 committed:
drop files

#不要なファイルの削除
testファイル
